src/api/data/cache/cache.go

```diff
@@ -0,0 +1,87 @@
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/helm/monocular/src/api/data"
+	"github.com/helm/monocular/src/api/data/helpers"
+	"github.com/helm/monocular/src/api/mocks"
+	"github.com/helm/monocular/src/api/swagger/models"
+)
+
+var mockCharts = mocks.NewMockCharts()
+
+type cachedCharts struct {
+	knownRepos []map[string]string
+	allCharts  map[string][]*models.ChartVersion
+	rwm        *sync.RWMutex
```

Comment:
this will be fine as-is if you change to a `sync.RWMutex`, but might wanna leave it a pointer as a defensive measure.

if you accidentally pass an instance of this struct by value like this:

```go
func (c cachedCharts) doSomething() {...}
```

your code will make a copy of the `sync.RWMutex` and render the mutex semantically incapable of doing mutual exclusion as you'd expect (`go vet` would complain too). here's a trivial example: https://play.golang.org/p/dendtURMEP

n.b. - there's no technical reason to remove the pointer right now, and the compiler would likely also complain for other reasons if you added a method without a pointer receiver. I'm being paranoid about this because it's a concurrency issue